src/meta/src/hummock/manager/mod.rs

```diff
@@ -1404,8 +1429,203 @@ where
+        for compaction_group in compaction_groups {
+            self.try_send_compaction_request(compaction_group)?;
+        }
+        let (schedule_ok, task_num) = self.poll_compaction_schedule_status(old_size).await;
+        let (compaction_ok, cur_version) = self
+            .poll_compaction_tasks_status(schedule_ok, task_num, &old_version)
+            .await;
+
+        tracing::info!(
+            "Compaction schedule_ok {}, task_num {} compaction_ok {}",
+            schedule_ok,
+            task_num,
+            compaction_ok,
+        );
+
+        if !schedule_ok && !compaction_ok {
+            return Ok((old_version.id, old_version.max_committed_epoch));
+        }
+        Ok((cur_version.id, cur_version.max_committed_epoch))
+    }
+
+    /// Poll the compaction task assignment to aware whether scheduling is success.
+    /// Returns (whether scheduling is success, number of tasks)
+    async fn poll_compaction_schedule_status(&self, old_size: usize) -> (bool, usize) {
+        let poll_timeout = Duration::from_secs(2);
+        let poll_interval = Duration::from_millis(20);
+        let mut poll_duration_cnt = Duration::from_millis(0);
+        let mut new_size = self.get_compact_task_assignment_size().await;
+        let mut schedule_ok = false;
+        loop {
+            if new_size > old_size {
+                schedule_ok = true;
+                break;
+            }
+
+            if poll_duration_cnt >= poll_timeout {
+                break;
+            }
+            tokio::time::sleep(poll_interval).await;
+            poll_duration_cnt += poll_interval;
+            new_size = self.get_compact_task_assignment_size().await;
+        }
+        (schedule_ok, new_size - old_size)
+    }
+
+    async fn poll_compaction_tasks_status(
+        &self,
+        schedule_ok: bool,
+        task_num: usize,
+        old_version: &HummockVersion,
+    ) -> (bool, HummockVersion) {
+        // Polls current version to check whether its id become large,
+        // which means compaction tasks have finished
+        let poll_timeout = if schedule_ok {
+            Duration::from_secs(120)
+        } else {
+            Duration::from_secs(5)
+        };
+        let poll_interval = Duration::from_millis(50);
+        let mut duration_cnt = Duration::from_millis(0);
+        let mut compaction_ok = false;
+        let mut cur_version = self.get_current_version().await;
+        loop {
+            if (cur_version.id > old_version.id)
+                && (cur_version.id - old_version.id >= task_num as u64)
+            {
+                tracing::info!("Collected all of compact tasks");
+                compaction_ok = true;
+                break;
+            }
+            if duration_cnt >= poll_timeout {
+                break;
+            }
+            tokio::time::sleep(poll_interval).await;
+            duration_cnt += poll_interval;
+            cur_version = self.get_current_version().await;
+        }
```

Comment:
```suggestion
        loop {
            let cur_version = self.get_current_version().await;
            if (cur_version.id > old_version.id)
                && (cur_version.id - old_version.id >= task_num as u64)
            {
                tracing::info!("Collected all of compact tasks");
                compaction_ok = true;
                break;
            }
            if duration_cnt >= poll_timeout {
                break;
            }
            tokio::time::sleep(poll_interval).await;
            duration_cnt += poll_interval;
        }
```

Reply:
`cur_version` should be defined out of the loop. Because it also needs to be returned from the function.